File: LeetCode/53 - Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
      int maxSum = nums[0];
      int runningSum = 0;
      for (int num: nums) {
        if (runningSum < 0) runningSum = 0;
        runningSum += num;
        maxSum = max(maxSum, runningSum);
      }
      return maxSum;
    }
};

File: LeetCode/53 - Maximum Subarray.py
class Solution:
  def maxSubArray(self, nums: List[int]) -> int:
    maxSum = nums[0]
    runningSum = 0
    for num in nums:
      if runningSum < 0: runningSum = 0
      runningSum += num
      maxSum = max(maxSum, runningSum)
    return maxSum
